lib/services/local_storage/local_storage_service.dart

```diff
@@ -34,9 +39,31 @@ class LocalStorageServiceImpl implements LocalStorageService {
     return Mapper.fromJson(json).toObject<T>();
   }
 
+  @override
+  Future<List<T>?> getListObject<T extends Mappable>(String key) async {
+    final SharedPreferences prefs = await SharedPreferences.getInstance();
+    final jsonString = prefs.getString(key);
+    if (jsonString == null) {
+      return null;
+    }
+
+    final json = jsonDecode(jsonString) as List<dynamic>;
+    return json
+        .map((e) => Mapper.fromJson(e as Map<String, dynamic>).toObject<T>()!)
+        .toList();
+  }
+
+  @override
+  Future<void> setListObject<T extends Mappable>(List<T> list,
+      {required String key}) async {
+    final SharedPreferences prefs = await SharedPreferences.getInstance();
+    final jsonList = list.map((e) => e.toJson()).toList();
+    await prefs.setString(key, jsonEncode(jsonList));
+  }
+
   @override
   Future<void> remove(String key) async {
     final SharedPreferences prefs = await SharedPreferences.getInstance();
-    prefs.remove(key);
+    await prefs.remove(key);
```

Comment:
🤔  I think `await` here could be a bit redundant? 🤔  on removal, we actually don't usually plan for checking the result right? perhaps just `void remove(String key)` ? 🤔

Reply:
@sleepylee I think it necessary to prevent this case
```dart
remove("test")
get("test") // it still available
```

Reply:
technically, you're right, it is async, so when testing calling 1-after-1, it can be too fast it didn't update correctly. Doing it right now doesn't harm much it's just making this become synchronously waiting for this result to arrive 🤔  anyway. But I'm not sure if the current approach will fix tho, because if you're calling `remove` then `get` - it will not respect your async order anyway. You will have to do something like: `remove().then { get() }` to make it work in order right? 🤔

Reply:
@sleepylee I will try to do that, it looks cleaner 😁 
```dart
Future<String?> removeThenGet() async {
 await remove("test")
 return get("test")
}
```

Reply:
Btw, I feel safer here if we leave a Future there, so the other will know it not remove immediately. 
You have to wait! 😁